<a href="https://colab.research.google.com/github/KristineYW/DS-Unit-2-Applied-Modeling/blob/master/Kristine_Wang_LS_DS_231_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Lambda School Data Science

*Unit 2, Sprint 3, Module 1*

---


# Define ML problems

You will use your portfolio project dataset for all assignments this sprint.

## Assignment

Complete these tasks for your project, and document your decisions.

- [ ] Choose your target. Which column in your tabular dataset will you predict?
- [ ] Is your problem regression or classification?
- [ ] How is your target distributed?
    - Classification: How many classes? Are the classes imbalanced?
    - Regression: Is the target right-skewed? If so, you may want to log transform the target.
- [ ] Choose your evaluation metric(s).
    - Classification: Is your majority class frequency >= 50% and < 70% ? If so, you can just use accuracy if you want. Outside that range, accuracy could be misleading. What evaluation metric will you choose, in addition to or instead of accuracy?
    - Regression: Will you use mean absolute error, root mean squared error, R^2, or other regression metrics?
- [ ] Choose which observations you will use to train, validate, and test your model.
    - Are some observations outliers? Will you exclude them?
    - Will you do a random split or a time-based split?
- [ ] Begin to clean and explore your data.
- [ ] Begin to choose which features, if any, to exclude. Would some features "leak" future information?

If you haven't found a dataset yet, do that today. [Review requirements for your portfolio project](https://lambdaschool.github.io/ds/unit2) and choose your dataset.

Some students worry, ***what if my model isn't “good”?*** Then, [produce a detailed tribute to your wrongness. That is science!](https://twitter.com/nathanwpyle/status/1176860147223867393)

In [0]:
# Import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [14]:
# Read in CSV file 

sc = pd.read_csv('https://raw.githubusercontent.com/KristineYW/DS-Unit-2-Applied-Modeling/master/SCDB_2019_01_caseCentered_Vote.csv', encoding = "ISO-8859-1")

sc.head()

caseId     docketId    caseIssuesId  ... splitVote majVotes  minVotes
0  1946-001  1946-001-01  1946-001-01-01  ...         1        8         1
1  1946-002  1946-002-01  1946-002-01-01  ...         1        6         3
2  1946-002  1946-002-02  1946-002-02-01  ...         1        6         3
3  1946-002  1946-002-03  1946-002-03-01  ...         1        6         3
4  1946-002  1946-002-04  1946-002-04-01  ...         1        6         3

[5 rows x 53 columns]

In [12]:
# Check out the profile 

from pandas_profiling import ProfileReport

profile = ProfileReport(sc)

profile.to_file(output_file='output.html')

TypeError: ignored

In [15]:
sc['partyWinning'].value_counts(normalize=True)

1.0    0.620302
0.0    0.379402
2.0    0.000296
Name: partyWinning, dtype: float64

#### Choose your target. Which column in your tabular dataset will you predict?

My target will be the column WIN, which is the winning party to the case. 

#### Is your problem regression or classification?

My problem is a classification-type.
 
#### How is your target distributed?

#### Classification: How many classes? Are the classes imbalanced?

There will be 2 classes

#### Regression: Is the target right-skewed? If so, you may want to log transform the target.

#### Choose your evaluation metric(s). Classification: Is your majority class frequency >= 50% and < 70% ? If so, you can just use accuracy if you want. Outside that range, accuracy could be misleading. What evaluation metric will you choose, in addition to or instead of accuracy?

Yes, my majority class frequency is ab out 62%, which is between 50-70%, so I will use accuracy as my evaluation metric.

#### Choose which observations you will use to train, validate, and test your model.

I will use the date range of 1946-1990 to train my model, then use 1990-2005 to validate, and 2005-2019 to test my model. 

#### Are some observations outliers? Will you exclude them?

I don't believe there are any outliers. 

#### Will you do a random split or a time-based split?

I will use a time-based split for my training and test data.

#### Begin to clean and explore your data.

#### Begin to choose which features, if any, to exclude. Would some features "leak" future information?